From 5e90383ae4ed855d5afe588055c43952314947c7 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:00:00 -0400
Subject: [fs] btrfs: fix missing skinny metadata check in scrub_stripe()

Message-id: <1407179109-13714-114-git-send-email-sandeen@redhat.com>
Patchwork-id: 87072
O-Subject: [RHEL7.1 PATCH 113/422] Btrfs: fix missing skinny metadata check in scrub_stripe()
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Wang Shilong <wangsl.fnst@cn.fujitsu.com>

commit 7c76edb77c23db673a83793686b4a53e2eec4de4
Author: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
Date:   Sun Jan 12 21:38:32 2014 +0800

    Btrfs: fix missing skinny metadata check in scrub_stripe()

    Check if we support skinny metadata firstly and fix to use
    right type to search.

    Signed-off-by: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/scrub.c b/fs/btrfs/scrub.c
index 7806e2c..e0677e4 100644
--- a/fs/btrfs/scrub.c
+++ b/fs/btrfs/scrub.c
@@ -2373,8 +2373,11 @@ static noinline_for_stack int scrub_stripe(struct scrub_ctx *sctx,
    scrub_blocked_if_needed(fs_info);
   }
 
+  if (btrfs_fs_incompat(fs_info, SKINNY_METADATA))
+   key.type = BTRFS_METADATA_ITEM_KEY;
+  else
+   key.type = BTRFS_EXTENT_ITEM_KEY;
   key.objectid = logical;
-  key.type = BTRFS_EXTENT_ITEM_KEY;
   key.offset = (u64)-1;
 
   ret = btrfs_search_slot(NULL, root, &key, path, 0, 0);
-- 
1.7.1